src/main/java/os/failsafe/executor/PersistentQueue.java

```diff
@@ -46,7 +46,7 @@ List<Task> peekAndLock(Set<String> processableTasks, int limit) {
                 return Collections.emptyList();
             }
 
-            List<Task> locked = taskRepository.lock(connection, nextTasksToLock);
+            List<Task> locked = taskRepository.lock(connection, nextTasksToLock, null);
```

Comment:
I think, if I am right, we are missing a test for this. There is a test method in PersistentQueueShould#peek_and_lock_next_task. Use a concrete value as nodeId and verify for this concrete value (instead of any()). This test should fail, if I am right :)